### DISCUSSION TITLE
How to fetch output of concourse job through HTTP API

### DISCUSSION THREAD

**Discussion by kalrashubham49** [concourse/concourse#6302] [Help & Support]
Hello,

I have integrated concourse HTTP API's for creating , unpausing and triggering build for the pipelines, however i am stuck how to get response of job in my pipeline.

My Operation succeeded , and i want to fetch response for my my pipeline job execution (not status) , but i am not able to find api in routes.go for  that , can you please help me with this.

<img width="1247" alt="Screenshot 2020-11-20 at 5 03 48 PM" src="https://user-images.githubusercontent.com/13597940/99796044-d07b8280-2b52-11eb-92a3-ed0273305a90.png">

In the attached screenshot , i want to get the response of the job but when i am hitting the API for getting job i am getting response as :

{"id":53,"name":"test-http","pipeline_name":"apipipeline24","team_name":"main","next_build":null,"finished_build":{"id":62,"team_name":"main","name":"1.1","status":"succeeded","job_name":"test-http","api_url":"/api/v1/builds/62","pipeline_name":"apipipeline24","start_time":1605842533,"end_time":1605842534,"rerun_number":1,"rerun_of":{"id":61,"name":"1"}},"outputs":[{"name":"clair-communication","resource":"clair"}],"groups":null}

In output i am getting only name and resource.

**Comment by kalrashubham49** [comment]
Any help regarding this would be heartly appreciated.. :)

**Comment by andy-paine** [comment]
If you are looking for the logs the are output in the build then the endpoint is `BuildEvents` - `/api/v1/builds/:build_id/events`. You should be aware that the HTTP API for Concourse is not necessarily stable and the only _supported_ clients of it are https://github.com/concourse/concourse/tree/master/go-concourse and the Concourse web UI.

**Comment by kalrashubham49** [comment]
Hi Andy ,

Thanks for the response, I acknowledge API might not be stable as this API
is not working in Concourse 6.6.0, Just checked its implementation in
Concourse 6.6.0 Go Code, there is nothing which is returning the data.

func (s *Server) BuildEvents(build db.Build) http.Handler {
return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
streamDone := make(chan struct{})

go func() {
defer close(streamDone)

s.eventHandlerFactory(s.logger, build).ServeHTTP(w, r)
}()

select {
case <-streamDone:
}
})
}

**Reply by chenbh** [reply]
The `s.eventHandlerFactory(s.logger, build).ServeHTTP(w, r)` is returning the data: https://github.com/concourse/concourse/blob/master/atc/api/buildserver/eventhandler.go#L18
